src/utils/type.ts

```diff
@@ -100,6 +101,42 @@ export function assertReference(value: unknown): asserts value is Reference {
   }
 }
 
+export function assertReferenceOrEns(value: unknown): asserts value is ReferenceOrEns {
+  if (typeof value !== 'string') {
+    throw new TypeError('ReferenceOrEns has to be a string!')
+  }
+
+  if (isHexString(value)) {
+    assertReference(value)
+
+    return
+  }
+
+  /**
+   * a.asdf - VALID
+   * test.eth - VALID
+   * ADAM.ETH - VALID
+   * ADAM UHLIR.ETH - INVALID
+   * test.whatever.eth - VALID
+   * -adg.ets - INVALID
+   * adg-.ets - INVALID
+   * as-a.com - VALID
+   * ethswarm.org - VALID
+   * http://asdf.asf - INVALID
+   * řš+ýí.šě+ř.čě - VALID
+   * tsg.asg?asg - INVALID
+   * tsg.asg:1599 - INVALID
+   */
+  const DOMAIN_REGEX = /^(?:[^-.\/?:\s][^.\/?:\s]{0,62}\.)+[^.\/?:\s]{2,63}$/
```

Comment:
Small improvement because what you have here actually matches `adg-.ets` as valid and also TLDs which start or end with `-`
```suggestion
  /**
   * a.asdf - VALID
   * test.eth - VALID
   * ADAM.ETH - VALID
   * ADAM UHLIR.ETH - INVALID
   * test.whatever.eth - VALID
   * -adg.ets - INVALID
   * adg-.ets - INVALID
   * as-a.com - VALID
   * ethswarm.org - VALID
   * http://asdf.asf - INVALID
   * řš+ýí.šě+ř.čě - VALID
   * tsg.asg?asg - INVALID
   * tsg.asg:1599 - INVALID
   * ethswarm.something- - INVALID
   * ethswarm.-something - INVALID
   * ethswarm.some-thing - VALID
   */
  const DOMAIN_REGEX = /^(?:(?!-)[^.\/?:\s]{1,63}(?<!-)\.)+(?!-)[^.\/?:\s]{2,63}(?<!-)$/
```